### PR TITLE
NASDF 0.1.3

### PR DESCRIPTION
**Pull request by aartaka**
Opening a PR to check it once more that tests pass.

### CODE REVIEW

**Comment by aartaka** [comment]
Interesting... `list-of` is not recognised as a type because it's a type with arguments. `documentation 'list-of 'type` returns the documentation just fine. Maybe use documentation as a fallback for `valid-type-p`?

File: .gitmodules
Comment (Ambrevar) [inline .gitmodules:1]: Why add cl-unicode?

**Comment by Ambrevar** [comment]
Isn't `valid-type-p` wrong anyways?

```lisp
(typep t 'number)
; => NIL
```

The `(documentation X 'type)` trick seems to work when type is documented, however it does not distinguish between undocumented types and non-types.

File: .gitmodules
Comment (aartaka) [reply .gitmodules:1]: `cl-interpol` depends on it. And `cl-interpol` is required by `lisp-unit2`.

**Comment by aartaka** [comment]
Finally, types with arguments are supported by NASDF, and the compilation test listing is much better:
```
; Errors:
((#<PACKAGE "NKEYMAPS">
  (CUA +CONTROL+ KEY-MODIFIERS KEY EMACS MAKE-KEY-REQUIRED-ARG +HYPER+
   KEY-STATUS KEYMAP EMPTY-MODIFIERS VI-INSERT DEFAULT EMPTY-VALUE KEYSCHEME-P
   +META+ KEY-VALUE MAKE-KEYMAP +SUPER+ KEYSCHEME KEYMAP-P KEYSCHEME-MAP-P
   +SHIFT+ KEY-CODE MODIFIER= VI-NORMAL COPY-KEY EMPTY-KEYSPEC))
 (#<PACKAGE "NKEYMAPS/CORE">
  (COPY-KEY MAKE-KEYMAP KEY-STATUS KEY-MODIFIERS KEYSCHEME KEYSCHEME-MAP-P
            KEYMAP-P KEYMAP MODIFIER= KEY KEYSCHEME-P KEY-CODE KEY-VALUE))
 (#<PACKAGE "NKEYMAPS/CONDITIONS">
  (EMPTY-KEYSPEC MAKE-KEY-REQUIRED-ARG EMPTY-VALUE EMPTY-MODIFIERS))
 (#<PACKAGE "NKEYMAPS/MODIFIER"> (+HYPER+ +CONTROL+ +SHIFT+ +META+ +SUPER+))
 (#<PACKAGE "NKEYMAPS/KEYSCHEME"> (DEFAULT EMACS VI-INSERT CUA VI-NORMAL)))
Found undocumented exported symbols in 5 packages.
```

@Ambrevar, willing to review the docstrings?

File: nasdf/compilation-tests.lisp [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                    ;; Parenscript macros don't have documentation.
+                    (and (find-package :parenscript)
+                         (gethash s (symbol-value (find-symbol "*MACRO-TOPLEVEL*" :parenscript)))))
+          (push s result))))))
+
+(flet ((list-offending-packages (package export-lister testing-for)
+         (let* ((package (find-package package)))
+           (delete nil
+                   (mapcar (lambda (package)
+                             (logger ";;; Testing ~a for ~a" package testing-for)
```
Comment (Ambrevar) [inline nasdf/compilation-tests.lisp:111]: These `;;;` are redundant with `logger`'s prefix which is `;`.

**Comment by Ambrevar** [comment]
Done with c295c9c8d230672baa06b4c90ab6dd4bed74c34c, however the struct accessor docstrings are still missing.

Do you know how to specify them?  Otherwise calling `(setf (documentation ...))` should work.

**Comment by aartaka** [comment]
> Done with [c295c9c](https://github.com/atlas-engineer/nkeymaps/commit/c295c9c8d230672baa06b4c90ab6dd4bed74c34c), however the struct accessor docstrings are still missing.
> 
> Do you know how to specify them? Otherwise calling `(setf (documentation ...))` should work.

I guess that's why Shinmera's projects often have a separate file with dozens of `(setf (documentation ..))` in them.

> EDIT: Better: NASDF should be able to skip them. Can we detect defstruct accessors?

Maybe ignore them, after all? That's why there's `:undocumented-symbols-to-ignore`: we can't cover all the cases, especially with slots and objects.

**Comment by Ambrevar** [comment]
I recall there was a need for structs, I think because of Fset.

**Comment by Ambrevar** [comment]
By the way, this version of NASDF seems now beyond 0.1.3 (compare to nclasses).  Shall we update it to 0.1.4 then?

**Comment by aartaka** [comment]
> ECL fails in the CI because it cannot find the type predicates, apparently.

ECL/Ubuntu is quite weird on CI in general, so I'd ignore it here.

**Pull request merged**